virttest/qemu_vm.py

```diff
@@ -1696,22 +1710,27 @@ def sort_key(dev):
         else:
             if smp == 0 or vcpu_sockets == 0:
                 vcpu_dies = vcpu_dies or 1
+                vcpu_clusters = vcpu_clusters or 1
                 vcpu_cores = vcpu_cores or 1
                 vcpu_threads = vcpu_threads or 1
                 if smp == 0:
                     vcpu_sockets = vcpu_sockets or 1
-                    smp = vcpu_cores * vcpu_threads * vcpu_dies * vcpu_sockets
+                    smp = vcpu_cores * vcpu_threads * vcpu_clusters * vcpu_dies * vcpu_sockets
                 else:
-                    vcpu_sockets = smp // (vcpu_cores * vcpu_threads * vcpu_dies) or 1
+                    vcpu_sockets = smp // (vcpu_cores * vcpu_threads * vcpu_clusters * vcpu_dies) or 1
             elif vcpu_dies == 0:
                 vcpu_cores = vcpu_cores or 1
                 vcpu_threads = vcpu_threads or 1
                 vcpu_dies = smp // (vcpu_sockets * vcpu_cores * vcpu_threads) or 1
+            elif vcpu_clusters == 0:
+                vcpu_cores = vcpu_cores or 1
+                vcpu_threads = vcpu_threads or 1
+                vcpu_clusters = smp // (vcpu_sockets * vcpu_cores * vcpu_threads) or 1
```

Comment:
Should consider `dies` and `clusters` together.

Please refer to https://gitlab.com/qemu-project/qemu/-/blob/master/hw/core/machine-smp.c#L71-159

Reply:
interesting!

refer to: https://gitlab.com/qemu-project/qemu/-/blob/master/hw/core/machine-smp.c#L99-112

For dies and clusters we do not need to calculate
So we should take them out of the if loop and assign them directly, what do you think?
@PaulYuuu 

like this:
```
if vcpu_dies >0:
    vcpu_dies = vcpu_dies
else:
    vcpu_dies = 1
if vcpu_clusters > 0:
    vcpu_clusters = vcpu_clusters
else:
    vcpu_clusters = 1

if vcpu_maxcpus != 0:
......
else:

```

Reply:
Due to the update of qemu's source code design, after discussion, we decided to update the calculation method synchronously in avocado-vt.